src/api/entities/Account/__tests__/Staking.ts

```diff
@@ -254,7 +257,10 @@ describe('Staking namespace', () => {
         returnValue: dsMockUtils.createMockRewardDestination('Stash'),
       });
 
-      jest.spyOn(staking, 'getController').mockResolvedValue(entityMockUtils.getAccountInstance());
+      jest
+        .spyOn(staking, 'getController')
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .mockResolvedValue(entityMockUtils.getAccountInstance() as any);
```

Comment:
I think, instead of adding so many `as any` assertions, you could have used directly - `new Account()` and added `jest.mock(
  '~/api/entities/Account',
  require('~/testUtils/mocks/entities').mockAccountModule('~/api/entities/Account')
);` to the top. Check if that works.

Reply:
This seems to be a bug or limitation in jest type inference. Both ways give this error

```ts
Argument of type 'Account' is not assignable to parameter of type 'UnsubCallback | Promise<UnsubCallback>'.ts(2345)
```

I think Jest has trouble with the overloaded methods, its inferring they should return an Unsub function only